extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/AbstractRequestWrapper.java

```diff
@@ -322,12 +322,12 @@ public Future<ServerWebSocket> toWebSocket() {
 
     @Override
     public Context context() {
-        return delegate.context();
+        return ((HttpServerRequestInternal) delegate).context();
```

Comment:
How doing that every time we get a context or a metric would improve things?

Reply:
Also, I would prefer if we only do this type of optimization if the improvement is proven. Making our code more fragile for something that might actually not improve things doesn't sound like a good thing to me.

Reply:
> How doing that every time we get a context or a metric would improve things?

That's why having knowledge which methods are most common and how they are used, makes difference.
The typo pollution agent (see https://github.com/RedHatPerf/type-pollution-agent/) and the linked JDK issue explain what scalability problem it can cause, but feel free to ping me on zulip's dev to get a more detailed explanation